### PR TITLE
vspec2yaml builder

### PR DESCRIPTION
**Pull request by crea7or**
As discussed in the issue #95 I have created YAML from vspec builder.

### CODE REVIEW

**Comment by erikbosch** [comment]
I would suggest to add it as a subtarget for travis_targets (yes it is still called so) in the Makefile (both here and in vss-repo) so that we it can be run automatically be Continuous Integration

**Comment by SebastianSchildt** [comment]
I tested it manually and it ran, generated a YAML file that is syntactically valid. It should be low risk, as it is using 90% of the same code as the other tools, so I would be fine with merging.

For the curious, this is how a snippet looks like

```yaml

Vehicle.Speed:
  type: sensor
  datatype: float
  unit: km/h
  description: Vehicle speed
  uuid: efe50798638d55fab18ab7d43cc490e9

Vehicle.TravelledDistance:
  type: sensor
  datatype: float
  unit: km
  description: Odometer reading, total distance travelled during the lifetime of the vehicle.
  uuid: 90be9d7b0ac15b75a83027ea3b73b65b

Vehicle.TripMeterReading:
  type: sensor
  datatype: float
  unit: km
  description: Current trip meter reading
  uuid: 81f51ebfe29c591190171d7b96e1c948

Vehicle.AmbientAirTemperature:
  type: sensor
  datatype: float
  unit: celsius
  description: Ambient air temperature outside the vehicle.
  uuid: 2ffcbc2e6ea75dd991e3ae80b29a1d85

Vehicle.IsMoving:
  type: sensor
  datatype: boolean
  description: Indicates whether the vehicle is stationary or moving
  uuid: db69549cc7375e919c2a2883b41cd19c

Vehicle.AverageSpeed:
  type: sensor
  datatype: float
  unit: km/h
  description: Average speed for the current trip
  uuid: 43a489636a665c3abb99b63174eb552b

Vehicle.Acceleration:
  type: branch
  description: Spatial acceleration
  uuid: 6c490e6a798c5abc8f0178ed6deae0a8

Vehicle.Acceleration.Longitudinal:
  type: sensor
  datatype: float
  unit: m/s^2
  description: Vehicle acceleration in X (longitudinal acceleration).
  uuid: 3d511fe7232b5841be311b37f322de5a


```

**Comment by SebastianSchildt** [comment]
Discussed 08/24: Please check and comment if you have an opinion
Open question: Is "vspec2yaml" a good name, as it might be confusing as vspec is also "yaml", maybe this should be called vspecflatten?

Also should it into contrib or as main tool. Please think about it & comment so we can discuss next week

**Review by gunnarx — CHANGES_REQUESTED** [review]
- The two commits can be squashed into one before we merge.
(and resolve comments from @SebastianSchildt above)

**Review by gunnarx — CHANGES_REQUESTED** [review]
- x (execution) flag missing on the vspec2yaml.py file

**Comment by crea7or** [comment]
@gunnarx it looks like these changes are not on my side, yes?

**Comment by danielwilms** [comment]
Summerizing:
1. let's try to find a different name
2. decide on if it goes to contrib or not

**Comment by gunnarx** [comment]
@crea7or wrote:
> @gunnarx it looks like these changes are not on my side, yes?

You need to at minimum set the -x flag and then recommit.
Squashing is theoretically possible to do by the person who is merging, but it is easier if you update the PR.

**Comment by crea7or** [comment]
> 1. let's try to find a different name (`vspecflatten`, `vspec2singleFile`, `vspec2yaml`, `vspec2singleYaml`)
>    favorite in the call was `vspec2singleYaml`. more suggestions?

Imho vspec2yaml is good. If vspec will be changes internally it will loose meaning. All other tools is named with this pattern and users should not think about why this file is named in the different way.
Whatever, I'll change it to any you choose.

**Review by SebastianSchildt — APPROVED** [review]
Looks good to me, so my vote is merge.

wrt. the name discussion I am fine with vspec2yaml (argument: vspec is not pure yaml that is pareseable in any "useful" way out of the with standard libs)) as well as with  any of the alternatives (argument: vspecs are written using yaml so "converting" to yaml  might feel confusing for some)

File: vspec2yaml.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+#
+# (c) 2016 Jaguar Land Rover
+# (c) 2021 Robert Bosch GmbH
+# (c) 2021 Pavel Sokolov (pavel.sokolov@gmail.com)
+#
+# All files and artifacts in this repository are licensed under the
+# provisions of the license provided by the LICENSE file in this repository.
+#
+#
+# Concert vspec file into flat YAML
```
Comment (gunnarx) [inline vspec2yaml.py:12]: Spelling: Convert

To clarify what it does how about expanding into this:

Convert all vspec input files to a single flat YAML file.

**Review by gunnarx — CHANGES_REQUESTED** [review]

File: vspec2yaml.py (hunk 1 above)
Comment (danielwilms) [reply vspec2yaml.py:12]: @crea7or: vspec2yaml is fine, after we voted in the call today. But please check the spelling as mentioned by @gunnarx. Thanks so much for the work and the change!

Comment (crea7or) [reply vspec2yaml.py]: @gunnarx @danielwilms done!

**Review by SebastianSchildt — APPROVED** [review]
Tested. (Still) works. Vote merge

File: vspec2yaml.py (hunk 1 above)
Comment (danielwilms) [reply vspec2yaml.py]: Thanks @crea7or!

**Comment by gunnarx** [comment]
Fixed conflicts resulting from changes that were merged to README before this one (see #97)
@crea7or Warning - since it is possible to force-push to update a PR, it means I now updated @crea7or/master on GitHub (because that is the "from" branch you used).

**Review by gunnarx — APPROVED** [review]

**Comment by gunnarx** [comment]
(fyi @erikbosch )
I messed this up slightly because I had a last minute local change that was not committed before merging:

For vspec2yaml the [table in the README](https://github.com/GENIVI/vss-tools/blob/aacda162ef21b0bea055585d8de57598fcf50e9f/README.md) now says "(Community Supported)" instead of simply "Community Supported".   I suppose it is something we can clean up in a later README update.

**Pull request merged**